### PR TITLE
Plugin: Enable service account based on plugin settings on init

### PR DESCRIPTION
**Pull request by gamab**
**What is this feature?**

With this PR, on start up, we set the state of the plugin service account (enabled/disabled) based on plugin settings.

In other words, if the plugin is disabled it won't be able to make calls to Grafana with its service account token.

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

https://github.com/grafana/grafana-enterprise/issues/5824

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/pluginsintegration/serviceregistration/serviceregistration.go [hunk 1]
```diff
@@ -2,31 +2,40 @@ package serviceregistration
...
-		os: os,
+		reg:         reg,
+		settingsSvc: settingsSvc,
 	}
 	return s
 }
 
 // RegisterExternalService is a simplified wrapper around SaveExternalService for the plugin use case.
 func (s *Service) RegisterExternalService(ctx context.Context, svcName string, svc *plugindef.ExternalServiceRegistration) (*auth.ExternalService, error) {
+	settings, err := s.settingsSvc.GetPluginSettingByPluginID(ctx, &pluginsettings.GetByPluginIDArgs{PluginID: svcName})
```
Comment (linoman) [inline pkg/services/pluginsintegration/serviceregistration/serviceregistration.go:29]: how intensive is this db query? will this impact the startup time of grafana itself?

**Review by linoman — COMMENTED** [review]
This looks great so far! Can't wait to see what's next.

I left a minor question on this.

File: pkg/services/pluginsintegration/serviceregistration/serviceregistration.go (hunk 1 above)
Comment (gamab) [reply pkg/services/pluginsintegration/serviceregistration/serviceregistration.go:29]: It's a `Get` query. It is IMO negligible compared to the cost of writing the service account, its role/permissions/assignment and its credentials.
Also we should consider that this runs only if a plugin requested a service account setup.

**Review by linoman — APPROVED** [review]

File: pkg/services/pluginsintegration/serviceregistration/serviceregistration.go (hunk 1 above)
Comment (andresmgot) [reply pkg/services/pluginsintegration/serviceregistration/serviceregistration.go:29]: This only works for `app` type plugins. `datasource` plugins should still be able to define an externalServiceRegistration (afaik) but those doesn't have settings. I guess you need to pass the plugin.Type here and only do this check if the type is `plugins.TypeApp`.

**Review by andresmgot — CHANGES_REQUESTED** [review]
I think we need to take into account the `datasource` plugin type here

**Review by IevaVasiljeva — APPROVED** [review]
This is really neat ✨ 

One question about enabling/disabling plugins - does this only happen during plugin registration (in `RegisterExternalService`)? Just want to check that there are no other places from which the service account status needs to be updated.

**Comment by gamab** [comment]
> I think we need to take into account the datasource plugin type here

Thanks for spotting this! :100:

**Review by andresmgot — APPROVED** [review]
LGTM now :+1:

**Pull request merged**